nft_ingester/src/consistency_bg_job.rs

```diff
@@ -0,0 +1,743 @@
+            continue;
+        };
+
+        let start = Instant::now();
+        let my_e_chksms = match we.list_epoch_checksums(grand_epoch, tree_pk).await {
+            Ok(v) => v,
+            Err(e) => {
+                tracing::error!("Error reading epochs from DB: {}", e.to_string());
+                return result;
+            }
+        };
+        metrics
+            .db_bubblegum_get_epochs_latency
+            .observe(start.elapsed().as_secs_f64());
+
+        let e_cmp_res = cmp(&my_e_chksms, &peer_e_chksms);
+        let epochs_to_check = e_cmp_res
+            .we_miss
+            .iter()
+            .chain(e_cmp_res.different.iter())
+            .map(|&a| (a.epoch, a.tree_pubkey))
+            .collect::<Vec<_>>();
+        for (epoch, tree_pubkey) in epochs_to_check {
+            let Ok(peer_changes) = peer
+                .list_epoch_changes(epoch, tree_pubkey, None, None)
+                .await
+            else {
+                continue;
+            };
+
+            let start = Instant::now();
+            let my_changes = match we.list_epoch_changes(epoch, tree_pubkey, None, None).await {
+                Ok(v) => v,
+                Err(e) => {
+                    tracing::error!("Error reading bubblegum changes from DB: {}", e.to_string());
+                    return result;
+                }
+            };
+            metrics
+                .db_bubblegum_get_changes_latency
+                .observe(start.elapsed().as_secs_f64());
+
+            let changes_cmp_res = cmp(&my_changes, &peer_changes);
+            result.extend(changes_cmp_res.we_miss.into_iter().map(|a| a.to_owned()));
+            // TODO: track different
+        }
+    }
+
+    result
+}
+
+#[allow(clippy::while_let_on_iterator)]
+async fn handle_missing_bbgm_changes(
+    missing_accounts: HashMap<BbgmChangeRecord, HashSet<usize>>,
+    trusted_peers: Vec<String>,
+    force_reingestable_slot_processor: Arc<dyn BlockConsumer + Send + Sync>,
+) {
+    let mut clients: HashMap<usize, grpc::client::Client> = HashMap::new();
+
+    let mut already_fetched_slots = HashSet::<u64>::new();
+    for (change, peers) in missing_accounts {
+        if already_fetched_slots.contains(&change.slot) {
+            continue;
+        }
+        let mut it = peers.iter();
+        while let Some(peer_ind) = it.next() {
+            let client = if let Some(client) = clients.get_mut(peer_ind) {
+                client
+            } else {
+                let Ok(peer_client) =
+                    grpc::client::Client::connect_to_url(&trusted_peers[*peer_ind]).await
+                else {
+                    continue;
+                };
+                clients.insert(*peer_ind, peer_client);
+                clients.get_mut(peer_ind).unwrap()
+            };
+            if let Ok(block) = client
+                .get_block(change.slot, Option::<Arc<grpc::client::Client>>::None)
```

Comment:
We fetching the whole block, without indicating which trees we want to receive. In the future we need to add methods for syncing on concrete trees